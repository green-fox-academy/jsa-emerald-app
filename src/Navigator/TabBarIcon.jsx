import React from 'react';
import PropTypes from 'prop-types';
import { NavigationScreenPropType } from 'react-navigation';
import { MaterialIcons } from '@expo/vector-icons';
import AddButton from './AddButton';

const TabBarIcon = ({ navigation, focused, tintColor }) => {
  const { routeName } = navigation.state;
  let iconName; let IconComponent;
  switch (routeName) {
    case 'Stats':
      IconComponent = MaterialIcons;
      iconName = `pie-chart${focused ? '' : '-outlined'}`;
      return <IconComponent name={iconName} size={25} color={tintColor} />;
    case 'Create':
      return <AddButton />;
    case 'Me':
      IconComponent = MaterialIcons;
      iconName = `person${focused ? '' : '-outline'}`;
      return <IconComponent name={iconName} size={25} color={tintColor} />;
    default:
      break;
  }
};

TabBarIcon.propTypes = {
  focused: PropTypes.bool,
  tintColor: PropTypes.string,
  navigation: PropTypes.shape(NavigationScreenPropType),
};

TabBarIcon.defaultProps = {
  focused: '',
  tintColor: '',
  navigation: {},
};

export default TabBarIcon;
