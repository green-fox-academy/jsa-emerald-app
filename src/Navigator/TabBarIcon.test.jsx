import React from 'react';
import * as Font from 'expo-font';
import renderer from 'react-test-renderer';
import { MaterialCommunityIcons, MaterialIcons } from '@expo/vector-icons';

import TabBarIcon from './TabBarIcon';

beforeAll(async () => {
  await Font.loadAsync(MaterialIcons.font);
  await Font.loadAsync(MaterialCommunityIcons.font);
  await Font.loadAsync(MaterialIcons.font);
});

describe('<TabBarIcon />', () => {
  it('render Stats', () => {
    const tree = renderer.create(<TabBarIcon
      navigation={{ state: { routeName: 'Stats' } }}
      focused
      tintColor="#5C6BC0"
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render Create', () => {
    const tree = renderer.create(<TabBarIcon
      navigation={{ state: { routeName: 'Create' } }}
      focused
      tintColor="#5C6BC0"
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render Me', () => {
    const tree = renderer.create(<TabBarIcon
      navigation={{ state: { routeName: 'Me' } }}
      focused
      tintColor="#5C6BC0"
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
