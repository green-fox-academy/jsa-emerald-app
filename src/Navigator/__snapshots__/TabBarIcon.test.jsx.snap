// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabBarIcon /> render Create 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "#5C6BC0",
        "fontSize": 25,
      },
      undefined,
      Object {
        "fontFamily": "material-community",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`<TabBarIcon /> render Me 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "#5C6BC0",
        "fontSize": 25,
      },
      undefined,
      Object {
        "fontFamily": "material",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;

exports[`<TabBarIcon /> render Trans 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "#5C6BC0",
        "fontSize": 25,
      },
      undefined,
      Object {
        "fontFamily": "material-community",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;
